src/api.c

```diff
@@ -536,7 +536,20 @@ static struct nmd_frame *ret_frame(struct nmd_ctx *s, AVFrame *frame, int status
 
 end:
     END_FUNC(MAX_SYNC_OP_TIME);
-    return ret;
+
+    *framep = ret;
+
+    int retcode;
+    if (ret)
+        retcode = NMD_RETCODE_NEWFRAME;
+    else if (status >= 0)
```

Comment:
I guess it could be `status == 0`, what do you think ?

Reply:
I'm not confident in FFmpeg code returns for every decoder etc so I'd rather keep it that way